lib/worker.js

```diff
@@ -90,14 +92,29 @@ Worker.prototype.run = function () {
   };
   return Promise.resolve().bind(this)
     .then(function runTheTask () {
-      var attemptData = { attempt: this.attempt++ };
+      var attemptData = {
+        attempt: this.attempt++,
+        timeout: this.msTimeout
+      };
       log.info(merge(jobAndQueueData, attemptData), 'Running task');
-      return this.task(this.job);
+      return Promise.resolve().bind(this)
+        .then(function () {
+          return Promise.resolve().bind(this)
+            .cancellable()
+            .then(function () { return this.task(this.job); });
+        })
+        .timeout(this.msTimeout);
```

Comment:
this has been changed now. If a user specifies no timeout (or a timeout of 0), it will not set a timeout on the promise.